src/MethodProxies-Tests/MpConcurrencyClass.class.st

```diff
@@ -0,0 +1,39 @@
+Class {
+	#name : #MpConcurrencyClass,
+	#superclass : #Object,
+	#instVars : [
+		'concurrencySemaphore'
+	],
+	#category : #'MethodProxies-Tests'
+}
+
+{ #category : #accessing }
+MpConcurrencyClass >> concurrencySemaphore [
+
+	^ concurrencySemaphore
+]
+
+{ #category : #accessing }
+MpConcurrencyClass >> concurrencySemaphore: anObject [
+
+	concurrencySemaphore := anObject
+]
+
+{ #category : #debugging }
+MpConcurrencyClass >> methodProcess1 [
+	self trigger1.
```

Comment:
Maybe some comments here could clarify the situation.

1st) We are in a base method, so here we can be instrumented.
2nd) After calling trigger1, the second process is meant to resume, right? There is then a window between the trigger and the signal that could cause break everything?

Why do we have a singnal following a wait?